model/task_history.go

```diff
@@ -335,7 +333,9 @@ func (iter *taskHistoryIterator) GetChunk(v *Version, numBefore, numAfter int, i
 		{"$group": groupStage},
 		{"$sort": bson.M{task.RevisionOrderNumberKey: -1}},
 	}
-	agg := database.C(task.Collection).Pipe(pipeline)
+	c := database.C(task.Collection)
+	c.SetMaxTime(time.Minute)
```

Comment:
This seems like the wrong place to set max time, since it has persistent side effects on future task queries. I think this either needs to be set in the environment (risky but maybe reasonable), or you need to restore the old value in a defer statement

Reply:
Doesn't each call to `DB()` and `C()` call create a new DB/collection wrapper instance? I thought setting max time here would only set it for this one collection wrapper and not persist the max time setting after this.

Reply:
ah I think you're right. To make this clearer can we instead pass through the max time as part of the AggregateOptions instead of setting it on the collection in the wrapper?